test/unit/dynamic-import.html

```diff
@@ -94,36 +71,117 @@
-            done();
+            setTimeout(function() {
+              errorSpy.reset();
+              target.dispatchEvent(new Event('error'));
+              assert.isFalse(errorSpy.called, 'doesn\'t trigger the error event listener');
+              done();
+            });
           });
-        });
+          return errorSpy;
+        }
 
-        Polymer.Base.importHref('dynamic-imports/async-import.html', loadSpy, errorSpy, true);
+        Polymer.Base.importHref('dynamic-imports/async-import-invalid.html', loadSpy, createErrorSpy(), true);
       });
 
-      test('importHref does not leak event listeners on error', function(done) {
-        var loadSpy = sinon.spy();
-        var errorSpy = sinon.spy(function(e) {
-          var target = e.target;
-          target.dispatchEvent(new Event('load'));
-          assert.isFalse(loadSpy.called, 'doesn\'t trigger the load event listener');
-          setTimeout(function() {
-            errorSpy.reset();
-            target.dispatchEvent(new Event('error'));
-            assert.isFalse(errorSpy.called, 'doesn\'t trigger the error event listener');
+      test('trigger load event of the second request of a resource when loading was not complete yet', function(done) {
+        var url = 'dynamic-imports/dynamic-notready.html'
+        Polymer.Base.importHref(url);
+        Polymer.Base.importHref(url, function() {
+          done();
+        });
+      });
+
+      test('trigger load event of the second request of a resource when loading was already completed', function(done) {
+        var url = 'dynamic-imports/dynamic-lazy.html';
+        var imprt = Polymer.Base.importHref(url);
+
+        var onLoad = function() {
+          imprt.removeEventListener('load', onLoad);
+          Polymer.Base.importHref(url, function() {
             done();
           });
+        }
+
+        imprt.addEventListener('load', onLoad);        
+      });
+
+      test('triggers the callback only once even when the element was requested again later', function(done) {
+        var url = 'dynamic-imports/dynamic-once.html';
+        var count = 0;
+        Polymer.Base.importHref(url, function() {
+          count++;
+          if(count === 1) {
+            Polymer.Base.importHref(url, function() {
+              setTimeout(function() {
+                assert.equal(count, 1, 'the first callback should only be called once');
+                done();
+              }, 0);
+            });
+          }
+        });
+      });
+
+      test('importHref does not cache failed results', function(done) {
+
+        var a, b;
+
+        a = Polymer.Base.importHref('does_not_exist.html', function() {
+          throw new Error('Load handler should not be called.');
+        }, function() {
+          b = Polymer.Base.importHref('does_not_exist.html', function() {
+            throw new Error('Load handler should still not be called.');
```

Comment:
Unless https://github.com/webcomponents/webcomponentsjs/pull/655 is merged this test fill fail on browsers without native html imports support (e.g. Firefox on Travis CI).